### PR TITLE
Simplifes OpInfo test matrix to reduce test time

### PR DESCRIPTION
**Pull request by mruberry**
This PR:

- Updates the structure of the SampleInput class to require the "input" attribute be a tensor
- Limits unary ufuncs to test only the uint8, long, float16, bfloat16, float and cfloat dtypes by default
- Limits variant testing to the float dtype
- Removes test_variant_consistency from test_unary_ufuncs.py since it's now redundant with variant testing in test_ops.py
- Adds backwards supported testing to clarify failures that were coming from variant testing

This should decrease test e2e time.

### CODE REVIEW

**Comment by mattip** [comment]
this PR has merge conflicts

**Comment by mruberry** [comment]
> this PR has merge conflicts

Yep; I'll update it soon.

**Comment by mattip** [comment]
Nice, this seems to have significantly sped up CI (comparing the far right column of [this PR's run](https://app.circleci.com/pipelines/github/pytorch/pytorch/280938/workflows/1b6051a6-0538-4441-8d0d-1a9e224e0132) against a random other [PR run](https://app.circleci.com/pipelines/github/pytorch/pytorch/280938/workflows/1b6051a6-0538-4441-8d0d-1a9e224e0132)

File: test/test_unary_ufuncs.py [hunk 1]
```diff
@@ -58,7 +58,7 @@
 _large_float_vals = _large_float16_vals + (-4988429.2, 4988429.2, -1e20, 1e20)
 _float_extremals = (float('inf'), float('-inf'), float('nan'))
 _medium_length = 812
-_large_size = (1029, 917)
+_large_size = (519, 917)
```
Comment (mruberry) [inline test/test_unary_ufuncs.py:61]: Revert this

File: test/test_ops.py [hunk 2]
```diff
@@ -180,6 +180,11 @@ def test_inplace_gradgrad(self, device, dtype, op):
 class TestCommon(JitCommonTestCase):
     exact_dtype = True
 
+    # variant testing is only done with torch.float to avoid excessive
```
Comment (mruberry) [inline test/test_ops.py:183]: Extend supported dtype testing to include backward testing and run on all sample inputs
also extend unsupported dtype testing to run on all sample inputs
measure total performance change

**Review by ngimel — APPROVED** [review]

**Comment by imaginary-person** [comment]
I don't know where I should ask this question, but does the failure of the test `test_supported_backward` (added in this PR) mean that backward is unsupported for that op?

While porting `OpInfo`s for some operators, it might be discovered that `test_supported_backward` for a dtype  fails due to some unsupported op for the `dtype`. For instance, `log_vml_cpu` does not support `float16` on CPU, so this test fails for `pow`.
When `test_variant_consistency_jit` test was still enabled on ops supporting `float16`,  it used to fail for `pow` because of this reason. But its eager counterpart (`test_variant_consistency_eager`)  used to pass (it used to fail when the inplace variant was tested, but it did used to progress that far, indicating that it worked on the main method).  Based on that, I inferred that `pow` wasn't _jittable_ for `float16`, but its eager mode worked.

But now the CI only runs `test_supported_backward` & it fails. So, should I assume that even the eager mode isn't working for `float` dtype for `pow`?

**Comment by mruberry** [comment]
> I don't know where I should ask this question, but does `test_variant_consistency_eager` subsume `test_unsupported_backward`?

test_variant_consistency_eager now runs only in float32, but test_unsupported_backward runs in every dtype that support autograd (like float16 and cfloat).

**Comment by imaginary-person** [comment]
> test_variant_consistency_eager now runs only in float32, but test_unsupported_backward runs in every dtype that support autograd (like float16 and cfloat).

Thanks for your response, @mruberry. 

I noticed that for the operator `pow`, `test_supported_backward` for `float16` fails due to the same cause (`log_vml_cpu` not being enabled for `float16` due to lack of complete AVX-2 vectorization support for `float16`) as
`test_variant_consistency_jit` (which has now been disabled for `float16`). 
But since `test_variant_consistency_eager` (also disabled now) used to pass for it (provided no sample inputs were provided that could cause any broadcasting semantics issue with `pow`'s inplace variant), I had assumed that `pow` wasn't _jittable_ for `float16`, but supported eager mode.

That's why I'm very confused as to what the failure of `test_supported_backward` signifies.

**Comment by mruberry** [comment]
> > test_variant_consistency_eager now runs only in float32, but test_unsupported_backward runs in every dtype that support autograd (like float16 and cfloat).
> 
> Thanks for your response, @mruberry.
> 
> I noticed that for the operator `pow`, `test_supported_backward` for `float16` fails due to the same cause (`log_vml_cpu` not being enabled for `float16` due to lack of complete AVX-2 vectorization support for `float16`) as
> `test_variant_consistency_jit` (which has now been disabled for `float16`).
> But since `test_variant_consistency_eager` (also disabled now) used to pass for it (provided no sample inputs were provided that could cause any broadcasting semantics issue with `pow`'s inplace variant), I had assumed that `pow` wasn't _jittable_ for `float16`, but supported eager mode.
> 
> That's why I'm very confused as to what the failure of `test_supported_backward` signifies, as I expected `pow` to support backward for `float16` in eager mode.

That's a really good question that required careful attention to detail.

The previous tests were very confusing because test_variant_consistency_eager used to have a try/except block around backward, which would allow backward to fail but the test to pass. This would allow ops like pow to pass the test, which is the intended behavior. The JIT version of the test, however, didn't have the same try/except block and so would report a failure. This led to the misleading signal you describe, where a reasonable person looking at the tests would think the operation wasn't supported by the jit. In fact, however, the op just didn't support backwards for a particular dtype.

I hope the new test structure better clarifies what's actually failing. For example, it now correctly indicates that pow's float16 issue was just that it doesn't support backward in float16, not that it isn't jittable.

Does that make sense?

**Comment by imaginary-person** [comment]
Thank you, @mruberry! Yes, the new test structure does indeed indicate what it should.

**Pull request closed without merging**